### PR TITLE
feat(release): Automated post-upgrade tasks by code generating upgrade handler and auto-increasing e2e version

### PR DESCRIPTION
**Pull request by RusAkh**
Closes: #2609 

## Brief Changelog

On release creation will be triggered GitHub action which will make a pull request with added new file for new version in app/upgrades and increase E2E_UPGRADE_VERSION in Makefile

https://github.com/RusAkh/osmosis-fork/pull/8 Here is a pull request from a fork which shows what exactly this action does

I am not sure if this should be documented or not, so making it draft for now. What do you think @p0mvn?

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? no
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? no
  - How is the feature or change documented? no

### CODE REVIEW

**Comment by RusAkh** [comment]
Also very important I have not yet checked tests for e2e 1) if they all pass 2) they test the new upgrade handler

File: .github/workflows/auto-update-upgrade.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
+name: On Release Auto Upgrade
```
Comment (p0mvn) [inline .github/workflows/auto-update-upgrade.yml:1]: This is awesome. Can you please add docs/comments directly in this file explaining what this CI job does? Excited to try it next upgrade!

**Review by p0mvn — APPROVED** [review]

File: .github/workflows/auto-update-upgrade.yml (hunk 1 above)
Comment (RusAkh) [reply .github/workflows/auto-update-upgrade.yml:1]: yes absolutely will!

**Comment by RusAkh** [comment]
@p0mvn docs added but please do not merge it yet I am not sure everything is good with e2e stuff, gotta check it tomorrow

**Comment by RusAkh** [comment]
hello @p0mvn the latest push also added a functionality to the script for changes in app.go file. Please check this pr https://github.com/RusAkh/osmosis-fork/pull/12 to view all the changes which are now pr'ed when release is created

**Comment by p0mvn** [comment]
You seem to have another more up to date PR related to this: https://github.com/osmosis-labs/osmosis/pull/3100

Can we close this one?

**Comment by RusAkh** [comment]
> You seem to have another more up to date PR related to this: #3100
> 
> Can we close this one?

yes, this one Is just to check e2e tests in CI

**Pull request closed without merging**